addons/token-exchange/blue_secret_controller.go

```diff
@@ -100,16 +108,23 @@ func (c *blueSecretTokenExchangeAgentController) sync(ctx context.Context, syncC
 		return nil
 	}
 
+	err = c.enableCSIOmapGenerator()
+	if err != nil {
+		return fmt.Errorf("failed to start omap-generator for rook: %v", err)
+	}
 	secret, err := getSecret(c.spokeSecretLister, name, namespace)
 	if err != nil {
 		return fmt.Errorf("failed to get the secret %q in namespace %q in managed cluster. Error %v", name, namespace, err)
 	}
 
 	sc, err := c.getStorageClusterFromRookSecret(secret)
 	if err != nil {
-		return fmt.Errorf("failed to get the storage cluster name from the sercet %q in namespace %q in managed cluster. Error %v", name, namespace, err)
+		return fmt.Errorf("failed to get the storage cluster name from the secret %q in namespace %q in managed cluster. Error %v", name, namespace, err)
+	}
+	err = c.enableMirroring(sc)
+	if err != nil {
+		return fmt.Errorf("failed to enable mirroring the storagecluster %q in namespace %q in managed cluster. Error %v", name, StorageNamespace, err)
```

Comment:
move this to MirrorPeer controller. Let this controller focus on secret exchange only.

Reply:
done